### PR TITLE
added discord bot file

### PR DESCRIPTION
**Pull request by boyuan12**
## Description

Added discord bot

Fixes # (issue)
N/A

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots
N/A

### CODE REVIEW

File: discord-bot/bot.py [hunk 1]
```diff
@@ -8,5 +22,79 @@ async def on_ready():
...
+                splitted_msg = str(message.content).split(f"<@!{bot_id}> ")[1].split(",")
+                amount = int(splitted_msg[1])
+                assignee = splitted_msg[2][4:-2]
+                description = splitted_msg[3]
+
+                print(assignee, message.author.id)
+                if str(assignee) == str(message.author.id):
+                    await message.channel.send("Can't add point to yourself!")
+                elif str(assignee) == str(bot_id):
+                    await message.channel.send("Thanks for giving me a point too!")
```
Comment (luiszugasti) [inline discord-bot/bot.py:53]: I like easter eggs

File: discord-bot/bot.py [hunk 2]
```diff
@@ -8,5 +22,79 @@ async def on_ready():
...
+            data = requests.get(GET_TOP_FELLOWS)
+            top_fellows = ""
+            for i in range(len(data.json()["data"])):
+                top_fellows += f"{i+1} - {data.json()['data'][i]['name']}\n"
+            await message.channel.send(top_fellows)
+
+        elif option == "add":
+            # only admin
+            if get(message.author.roles, name="admin"):
+                splitted_msg = str(message.content).split(f"<@!{bot_id}> ")[1].split(",")
```
Comment (luiszugasti) [inline discord-bot/bot.py:44]: Do we want to have a error handling if the user doesn't provide a correctly formed request? For example `add "I did a great job!" 10`

File: discord-bot/bot.py (hunk 1 above)
Comment (boyuan12) [reply discord-bot/bot.py:53]: me too

File: discord-bot/bot.py (hunk 2 above)
Comment (boyuan12) [reply discord-bot/bot.py:44]: Done

**Review by cqvu — APPROVED** [review]

**Pull request merged**